### PR TITLE
Harder Difficulty Scaling

### PR DESCRIPTION
**Pull request by DrParadox7**
Adds config for harder difficulty scaling:

If `difficultyScalingHungerOption` is true:
Peaceful = Hunger loss is reduced by 66%.
Easy = Hunger loss is reduced by 33%.
Normal = No Change.
Hard = **[Before]** No Change | **[Now]** Hunger loss is increased by 33%*.

*(requires `harderScalingHunger`).

### CODE REVIEW

File: src/main/java/iguanaman/hungeroverhaul/config/IguanaConfig.java [hunk 1]
```diff
@@ -241,7 +244,7 @@ public static <T> ConfigOption<T> putOptionInList(ConfigCategory category, Strin
     public static int respawnHungerDifficultyModifier;
     public static final ConfigOption<Boolean> disableHealingHungerDrainOption = addOption(CATEGORY_HUNGER, "disableHealingHungerDrain", true, false, "Disable the hunger drain when healing that was introduced in vanilla 1.6.2");
     public static boolean disableHealingHungerDrain;
-    public static final ConfigOption<Float> hungerLossRatePercentageOption = addOption(CATEGORY_HUNGER, "hungerLossRatePercentage", 4F/3F * 100F, 0F, null, 100F, "Speed up or slow down the rate that hunger drops (set to 0 to disable hunger loss)");
+    public static final ConfigOption<Float> hungerLossRatePercentageOption = addOption(CATEGORY_HUNGER, "hungerLossRatePercentage", 100F, 0F, null, 100F, "Speed up or slow down the rate that hunger drops (set to 0 to disable hunger loss)");
```
Comment (eigenraven) [inline src/main/java/iguanaman/hungeroverhaul/config/IguanaConfig.java]: Why change the default here from 133% to 100%?

Comment (DrParadox7) [reply src/main/java/iguanaman/hungeroverhaul/config/IguanaConfig.java]: To keep a similar level of difficulty in HARD while making the difficulty curve linear.

With this chart it should be easy to comprehend why the base config needs to be changed:
image
The dashed line shows if `hungerLossRatePercentage` is kept at default values.

### Why do so?
There are players who play the pack to enjoy themselves (NORMAL) and those who want to experience the c*** and b**** torture to challenge themselves (HARD).

Where before both had to deal with the same difficulty level this commit changes that.

This allows us to better tailor each experience as the food lasts longer for everyone else while a little less for those in HARD mode (3.33% shorter)

**Review by mitchej123 — CHANGES_REQUESTED** [review]
This mod is ARR.  Unless dream has succeeded in getting permission, we will need to use mixins or see if upstream will publish a new version.

**Comment by Dream-Master** [comment]
i can try again reach someone. No promise

**Comment by Dream-Master** [comment]
i close this now because i not reach anyone.

**Pull request closed without merging**